src/MoBi.Core/Snapshots/Mappers/ProjectMapper.cs

```diff
@@ -119,6 +127,47 @@ private async Task<ModelProject> mapToModel(SnapshotProject projectSnapshot, Pro
          }
       }
 
+      MoBiSimulation[] sims;
+      try
+      {
+         sims = await Task.WhenAll(tasks).ConfigureAwait(false);
+      }
+      catch (Exception ex)
+      {
+         _logger.AddException(ex);
+         sims = tasks
+            .Where(t => t.Status == TaskStatus.RanToCompletion)
+            .Select(t => t.Result)
+            .ToArray();
+      }
+
+      sims.Each(x => addSimulations(project, x));
+
+      if (simulationContext.Run && sims.Any())
+      {
+         var maxParallel = Math.Max(1, _userSettings.MaximumNumberOfCoresToUse);
+         using var gate = new SemaphoreSlim(maxParallel);
+
+         var runTasks = sims.Select(async sim =>
+         {
+            await gate.WaitAsync();
+            try
+            {
+               await _simulationRunner.RunSimulationAsync(sim);
+            }
+            catch (Exception ex)
+            {
+               _logger.AddException(ex);
+            }
+            finally
+            {
+               gate.Release();
+            }
+         });
+
+         await Task.WhenAll(runTasks);
+      }
```

Comment:
Were you able to investigate `IConcurrentSimulationRunner` from core to see if it can be used? It is very similar to this code.